src/snowflake/connector/cursor.py

```diff
@@ -778,8 +779,10 @@ def execute(
                 "sqlstate": self._sqlstate,
                 "sfqid": self._sfqid,
             }
+            null_in_non_nullable = err == "NULL result in a non-nullable column"
```

Comment:
Perhaps it's better to check `code` (which appears to be `'100072'` for this error) if that's how this works.